windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -148,47 +148,40 @@ func TestReceiveCommandStartStop(t *testing.T) {
 func TestInstall(t *testing.T) {
 	t.Parallel()
 
-	ctx, cancel := context.WithCancel(context.Background())
-	t.Cleanup(cancel)
-
-	fileServerAddr := mockRootfsFileServer(t, ctx)
-
-	emptyFileChecksum := "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
-	mockErrorChecksum := "afe55cda4210c2439b47c62c01039027522f7ed4abdb113972b3030b3359532a"
-	mockMismatchChecksum := "1234"
-
 	testCases := map[string]struct {
 		noCloudInit            bool
 		cloudInitWriteErr      bool
 		distroAlreadyInstalled bool
-		emptyDistroName        bool
+		distroName             interface{} // using interface{} to allow nil, modelling an optional value.
```

Comment:
I would rather use a pointer then. But in general, we do something else in all our projects:
```
"" is the default
"-" means we replace with the empty value, so empty string for a string
```

For instance https://github.com/canonical/ubuntu-pro-for-wsl/blob/main/common/i18n/i18n_test.go#L115

Reply:
Makes sense. Is there a reason for not using a ´switch` instead of if-elses since at least three different cases are expected? (`""`, `"-"` and anything else) ?

Reply:
I think a `switch` reads better:

```go

			if tc.appxDoesNotExist || tc.sendRootfsURL != "" {
				// WSLMock Install only accepts ubuntu-22.04
				switch tc.distroName {
				case "-":
					settings.name = ""
				case "":
					settings.name = wsltestutils.RandomDistroName(t)
				default:
					settings.name = tc.distroName
				}
			}
```

Reply:
It depends on the context, when we have 2, this is fine. I still like the pattern in general to set the default, then handle special case(s). If there is only one, this is a no-brainer, as here there are 2 + one default, this is debatable. Happy with your solution :)